cmd/subnetcmd/upgradecmd/apply.go

```diff
@@ -204,29 +200,151 @@ func applyLocalNetworkUpgrade(subnetName string, sc models.Sidecar) error {
 		ux.Logger.PrintToUser("The next upgrade will go into effect %s", time.Unix(nextUpgrade, 0).Local().Format(constants.TimeParseLayout))
 		ux.PrintTableEndpoints(clusterInfo)
 
-		// it seems all went well this far, now we try to write/update the lock file
-		// if this fails, we probably don't want to cause an error to the user?
-		// so we are silently failing, just write a log entry
-		wrapper := params.UpgradeConfig{
-			PrecompileUpgrades: precmpUpgrades,
+		return writeLockFile(precmpUpgrades, subnetName)
+	}
+
+	return errors.New("unexpected network size of zero nodes")
+}
+
+// applyPublicNetworkUpgrade applies an upgrade file to a locally running validator
+// for public networks (fuji, main)
+// the validation of the upgrade file has many things to consider:
+// * No upgrade file for <public net> can be found - do we copy the existing file in the prev stage?
```

Comment:
are this questions going to be permanent here?

Reply:
Until we have found a way to answer these questions conclusively, yes. This keeps history and is important documentation. Next week already we will remember just half of it.